docs(css): Fx116 - syntax updates for offset-path and ray() (#28348)
mdn/content: files/en-us/web/css/offset-path/index.md

### Values

> The `offset-path` takes as it's value an `<offset-path>` value, a `<coord-box>` value, or both or the `none` keyword. The `<offset-path>` value is a {{cssxref("ray","ray()")}}, a {{cssxref("url","url()")}}, or a [`<basic-shape>`](/en-US/docs/Web/CSS/basic-shape) value.

Reviewer: link  `<coord-box>` 

and `<offset-path>` if we have a page / section for it.

Author: Added link for `<coord-box>`.
`<offset-path>` is defined in this section itself.

- `none`

>   - : Specifies that the element does not follow any offset path. This is the default value. This value is equivalent of the element not having any [offset transform](/en-US/docs/Web/CSS/offset). The element's movement in this case is determined by its default position properties, such as {{cssxref("top")}} and {{cssxref("left")}}, instead of an offset path.

Author: I agree adhering to those styles. Fixed as suggested.

- `<offset-path>`

>   - : Specifies the geometrical offset path. This parameter is optional. If not specified, the default value of the path shape is `inset(0 round X)`, where `X` is the value of {{cssxref("border-radius")}} of the element that establishes the [containing block](/en-US/docs/Web/CSS/Containing_block). `<offset-path>` is specified as one of the following three values:

Author: Going with the first suggestion. Keeping the coord-box details here though because it is more about what happens when none of the `<offset-path>` values are specified

    - {{cssxref("ray","ray()")}}

>       - : Defines the path, which is a line segment, from an element's starting position and extending at the specified angle. This value accepts up to four parameters – an {{CSSxRef("angle")}}, an optional size value, the optional keyword `contain`, and an optional `at <position>`.

Author: Going with this:

> Defines a line starting at a set position, of a set length, and extending at the specified angle. The `ray()` function accepts up to four parameters – an {{CSSxRef("angle")}}, an optional size value, the optional keyword `contain`, and an optional `at <position>`.

Wow, didn't know about the `<size>` parameter in [radial-gradient](https://developer.mozilla.org/en-US/docs/Web/CSS/gradient/radial-gradient). Thanks! I guess the only similarities between `radial-gradient` and `ray()` are `<size>` and `<position>`.

    - {{cssxref("url","url()")}}

      - : Specifies the URL reference of an SVG shape element — `circle`, `ellipse`, `line`, `path`, `polygon`, `polyline`, or `rect`. The referenced shape's geometry is the equivalent offset path. If the URL does not reference a shape element (because it references a different element or resolves to a non-SVG document or doesn't resolve at all), the derived offset path is `path("M0,0")` (that is, a `<basic-shape>`).

    - [`<basic-shape>`](/en-US/docs/Web/CSS/basic-shape)

>       - : Specifies the offset path as the equivalent path of a [basic shape](/en-US/docs/Web/CSS/CSS_shapes/Basic_shapes) function such as [`circle()`](/en-US/docs/Web/CSS/basic-shape/circle), [`ellipse()`](/en-US/docs/Web/CSS/basic-shape/ellipse), [`inset()`](/en-US/docs/Web/CSS/basic-shape/inset), {{cssxref("path","path()")}}, or [`polygon()`](/en-US/docs/Web/CSS/basic-shape/polygon). For example, if the `<basic_shape>` is an `ellipse()` function, then the path is the outline of the circle/ellipse – it starts at the rightmost point of the circle/ellipse, and then is composed of four circular arcs, each comprising a quarter of the circle/ellipse, proceeding clockwise, ending with a segment-completing close path operation. If a `<basic-shape>` type accepts the `at <position>` parameter but the parameter is omitted, and the element has an {{cssxref("offset-position")}}, the `offset-position` value is used for the `at <position>` parameter. Otherwise, the `at <position>` parameter defaults as specified for each shape function.

Reviewer: * added CSS so readers realize it's not SVG.
* changed ellipse/circle to simply circle, to make it easier to parse 
* got rid of the 4 arcs because the fact it's divided into 4 arcs has no bearing on the outcome of the path.
* explained the default for `at <position>` (am I correct?) if committed, then added if ommitted AND there is offset position.... the default for `at <position>` needs to be added to the ellipse page.

>       - : Specifies the offset path as the equivalent path of a [basic shape](/en-US/docs/Web/CSS/CSS_shapes/Basic_shapes) function such as [`circle()`](/en-US/docs/Web/CSS/basic-shape/circle), [`ellipse()`](/en-US/docs/Web/CSS/basic-shape/ellipse), [`inset()`](/en-US/docs/Web/CSS/basic-shape/inset), {{cssxref("path","path()")}}, or [`polygon()`](/en-US/docs/Web/CSS/basic-shape/polygon). For example, if the `<basic_shape>` is an `ellipse()` function, then the path is the outline of the circle/ellipse – it starts at the rightmost point of the circle/ellipse, and then is composed of four circular arcs, each comprising a quarter of the circle/ellipse, proceeding clockwise, ending with a segment-completing close path operation. If a `<basic-shape>` type accepts the `at <position>` parameter but the parameter is omitted, and the element has an {{cssxref("offset-position")}}, the `offset-position` value is used for the `at <position>` parameter. Otherwise, the `at <position>` parameter defaults as specified for each shape function.

Author: Thanks. I've made the changes as suggested. I've now also added the `circle()` and `ellipse()` pages to this PR to add the default value for `<position>`.
To note that [circle and ellipse sections](https://developer.mozilla.org/en-US/docs/Web/CSS/basic-shape#ellipse) on the `<basic-shape>` already contained the default info.

> - {{cssxref("box_value","&lt;coord-box&gt;")}}

Reviewer: Reviewed to this point

> - {{cssxref("box_value","&lt;coord-box&gt;")}}

Author: Going with the second suggestion, thanks!

>   - : Specifies the size information of the reference box containing the path. The reference box is derived from the element that establishes the containing block for this element. This parameter is optional. If not specified, the default value is `border-box`. In SVG, the value is treated as `view-box`. If `ray()` or `<basic-shape>` is used to define the offset path, the `<coord-box>` value provides the reference box for the ray or the `<basic-shape>`, respectively. If `url()` is used to define the offset path, the `<coord-box>` value provides the viewport and user coordinate system for the shape element, with the origin (`0 0`) at the top left corner and size being `1px`.

Reviewer: do we have a glossary or similar explanation to add a link for  "reference box"

Author: Done. Added link to https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_shapes/Basic_shapes#the_reference_box.

## Examples

> ### Creating an offset-path using coord-box positioning

Reviewer: ```suggestion
### Creating an offset-path using box-edge positioning
```

> This example shows the how `<coord-box>` parameter of `offset-path` works in with {{cssxref("border-radius")}}.

Reviewer: ```suggestion
This example shows how the `<coord-box>` box-edge values of `offset-path` work with {{cssxref("border-radius")}}.
```

Author: Updated this slightly differently to:

> This example demonstrates using various `<coord-box>` values in the `offset-path` property.

The word 'box' next to each other affects readability, so keeping it simple with "`<coord-box>` values".

```html hidden
<div class="box blueBox"></div>
<div class="box redBox"></div>
<div class="box greenBox"></div>
```

```css
body {
  width: 300px;
  height: 200px;
  border-radius: 50px;
  border: dashed aqua;
  border-width: 25px;
  padding: 25px;
  margin: 50px;
}

.box {
  width: 40px;
  height: 20px;
  animation: move 8000ms infinite ease-in-out;
}

.blueBox {
  background-color: blue;
  offset-path: border-box;
  offset-distance: 5%;
}

.greenBox {
  background-color: green;
  offset-path: padding-box;
  offset-distance: 8%;
}

.redBox {
  background-color: red;
  offset-path: content-box;
  offset-distance: 12%;
}

@keyframes move {
  0%,
  20% {
    offset-distance: 0%;
  }
  80%,
  100% {
    offset-distance: 100%;
  }
}
```

> In this example, the margin, border, and padding have been purposely been given large values to demonstrate the placement of the blue, green, and red boxes on their respective `coord-box` edges: border-box, padding-box, and content-box.

Reviewer: ```suggestion
In this example, the margin, border, and padding have been purposely given large values to demonstrate the placement of the blue, green, and red boxes on their respective `<coord-box>` box edges: border-box, padding-box, and content-box.
```

> ![Box model showing the placement of the blue, red, and green boxes on the border-box, padding-box, and context-box, respectively, by using the coord-box value in the offset-path property.](offset-path-coord-box.png)

Reviewer: ```suggestion
![The blue rectangle sits on the outer edge of the border, the red box is on the outer edge of the content box, and the green box is on the inner border edge, which is the outer edge of the padding](offset-path-coord-box.png)
```

the alt should describe what one learns from the image.

Author: Thanks. Updated to follow the order of blue to red to green rectangles from outer to inner boxes. Also calling the colored shapes rectangles to avoid confusion with border box etc.

> The blue rectangle sits on the outer edge of the border box, the green rectangle is on the inner border edge, which is the outer edge of the padding box, and the red rectangle is on the outer edge of the content box.

#### Result

> {{EmbedLiveSample('Creating an offset-path using coord-box positioning', '100%', 400)}}

Reviewer: ```suggestion
{{EmbedLiveSample('Creating an offset-path using box-edge positioning', '100%', 400)}}
```

### Animating an element with offset-path

> The `offset-path` properties in the CSS code sample defines a path that is identical to the `<path>` element in the SVG. The path, as can be seen in the rendering of the SVG code, is a line drawing of a house with a chimney.

Reviewer: ```suggestion
The `offset-path` properties in the CSS code sample define a path that is identical to the `<path>` element in the SVG. The path of the SVG animation is a line drawing of a house with a chimney.
```
Some people will have animations turnd off or otherwise can't see the animation. So changing the wording a bit

Author: Thanks!
